compiler/src/iree/compiler/DispatchCreation/FuseHorizontalContractions.cpp

```diff
@@ -283,336 +252,243 @@ static std::optional<HorizontalFusionGroup> getHorizontalFusionGroupMembers(
-    SmallVector<int64_t> newTruncShape = {
-        static_cast<int64_t>(rhsValues.size())};
-    newTruncShape.append(baseTruncType.getShape().begin(),
-                         baseTruncType.getShape().end());
-    auto newTruncType =
-        RankedTensorType::get(newTruncShape, baseTruncType.getElementType());
-    SmallVector<utils::IteratorType> newTruncIteratorTypes = {
-        utils::IteratorType::parallel};
-    newTruncIteratorTypes.append(baseTruncOp.getIteratorTypesArray());
-
-    ArrayRef newTruncOperandsRef(newTruncOperands);
-    linalg::GenericOp newTruncOp = rewriter.create<linalg::GenericOp>(
-        loc, newTruncType,
-        newTruncOperandsRef.take_front(baseTruncOp.getNumDpsInputs()),
-        newTruncOperandsRef.take_back(baseTruncOp.getNumDpsInits()),
-        newTruncIndexingMaps, newTruncIteratorTypes);
-
-    rewriter.cloneRegionBefore(baseTruncOp->getRegion(0),
-                               newTruncOp.getRegion(),
-                               newTruncOp.getRegion().begin());
-
-    concatResultOp = cast<linalg::LinalgOp>(newTruncOp.getOperation());
+    // Append the RHS operands;
+    SmallVector<OpOperand *> ins = linalgOp.getDpsInputOperands();
+    llvm::append_range(
+        fusedIns,
+        llvm::map_range(ArrayRef<OpOperand *>(ins).drop_front(),
+                        [](OpOperand *operand) { return operand->get(); }));
+
+    // Append the Outs operands.
+    llvm::append_range(fusedOuts, llvm::map_range(linalgOp.getDpsInitsMutable(),
+                                                  [](OpOperand &operand) {
+                                                    return operand.get();
+                                                  }));
+
+    // Append the result types.
+    fusedResultTypes.append(linalgOp->result_type_begin(),
+                            linalgOp->result_type_end());
+
+    // Append the rhs indexing maps.
+    llvm::append_range(fusedInsIndexingMaps,
+                       ArrayRef<AffineMap>(opIndexingMaps)
+                           .slice(1, linalgOp.getNumDpsInputs() - 1));
+
+    // Append the outs indexing maps.
+    llvm::append_range(fusedOutsIndexingMaps,
+                       ArrayRef<AffineMap>(opIndexingMaps)
+                           .drop_front(linalgOp.getNumDpsInputs()));
   }
 
-  SmallVector<SmallVector<OpFoldResult>> concatResultShape;
-  if (failed(concatResultOp.reifyResultShapes(rewriter, concatResultShape))) {
-    return baseContractOp.emitOpError(
-        "failed to get shape of concatenated result op");
+  SmallVector<AffineMap> fusedIndexingMaps = std::move(fusedInsIndexingMaps);
+  fusedIndexingMaps.append(fusedOutsIndexingMaps);
+  auto fusedOp = rewriter.create<linalg::GenericOp>(
+      loc, fusedResultTypes, fusedIns, fusedOuts, fusedIndexingMaps,
+      fusedIteratorTypes, [](OpBuilder &, Location, ValueRange) {});
+
+  Block *fusedBody = fusedOp.getBlock();
+  auto rhsIndex = 0;
+  auto outsIndex = fusedOp.getNumDpsInputs();
```

Comment:
style nit: spell out the type.

> Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type.

https://google.github.io/styleguide/cppguide.html#Type_deduction

>  Use auto if and only if it makes the code more readable or easier to maintain.

https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable